source/core/parameters.cc

```diff
@@ -1607,6 +1653,79 @@ namespace Parameters
         variable = Variable::phase;
       if (vop == "temperature")
         variable = Variable::temperature;
+      // end WIP
+
+      // WIP Create map of refinement variables
+      const std::string        var_op    = prm.get("variables");
+      std::vector<std::string> var_vec   = split_input_variables(var_op);
+      const std::string        type_op   = prm.get("types");
+      std::vector<std::string> type_vec  = split_input_variables(type_op);
+      const std::string        coars_op  = prm.get("fractions coarsening");
+      std::vector<std::string> coars_vec = split_input_variables(coars_op);
+      const std::string        refin_op  = prm.get("fractions refinement");
+      std::vector<std::string> refin_vec = split_input_variables(refin_op);
+
+      // Checking that the sizes are coherent
+      Assert(type_vec.size() == var_vec.size(),
+             MultipleAdaptationSizeError("types",
+                                         type_vec.size(),
+                                         var_vec.size()));
+      Assert(coars_vec.size() == var_vec.size(),
+             MultipleAdaptationSizeError("fractions coarsening",
+                                         coars_vec.size(),
+                                         var_vec.size()));
+      Assert(refin_vec.size() == var_vec.size(),
+             MultipleAdaptationSizeError("fractions refinement",
+                                         refin_vec.size(),
+                                         var_vec.size()));
+
+      for (std::vector<int>::size_type i = 0; i != var_vec.size(); ++i)
+        {
+          // Parsing variable for this index
+          if (var_vec[i] == "velocity")
+            vars = Variable::velocity;
+          else if (var_vec[i] == "pressure")
+            vars = Variable::pressure;
+          else if (var_vec[i] == "phase")
+            vars = Variable::phase;
+          else if (var_vec[i] == "temperature")
+            vars = Variable::temperature;
+          else
+            {
+              std::cerr
+                << "  WARNING skipping unknown variable in 'subsection mesh adaptation':"
+                << std::endl
+                << var_vec[i] << std::endl;
```

Comment:
I would just throw here instead of giving a warning.

Reply:
Yup, throwing is a better option here.